### ISSUE TITLE
[DatePicker] Подсвечивать определенные даты в календаре DatePicker

### ISSUE THREAD

**Issue by eugenialobanova** [skbkontur/retail-ui#1134]
http://tech.skbkontur.ru/react-ui/#/Components/DatePicker
Хочется, чтобы была возможность подсвечивать определенные даты в календаре DatePicker.

Сценарий такой:
есть отчеты, которые лежат в Драйве2. За один день - один отчет. Но отчеты есть не за все дни. Хочу подсвечивать в календаре те дни, когда за которые отчет есть.

Более общий случай -  хочется, чтобы при определенном условии нужные даты в момент выбора были подсвечены.

**Comment by dzekh** [comment]
Необходима ли функциональность выбора тех дней в которых нет отчетов, может их можно просто блокировать, дизэйблить? 
Либо использовать для какого-то типа дней стиль выходных дней — красить красным?

**Comment by wKich** [comment]
Дизейблить сейчас можно только те дни, которые выходят за диапазон `minDate/maxDate`

Помечать же красным, конечно можно, но это больше подходит только как временное решение.

Сейчас не понятно как должна выглядеть подсветка, чтобы она не конфликтовала с подсветкой на hover.

**Comment by eugenialobanova** [comment]
> Необходима ли функциональность выбора тех дней в которых нет отчетов, может их можно просто блокировать, дизэйблить?
Да, дизэйблить дни без отчетов тоже бы подошло. Но, как я поняла из комментария ниже, сейчас так тоже не получится сделать. 
Возможно, кстати, дизэйблить дни без отчетов чем-то даже лучше.

**Comment by eugenialobanova** [comment]
Воспользовалась пока двумя свойствами: disabled для минимальной и максимальной даты и isHoliday. Получилось как-то так: https://mon.testkontur.ru/stamon/#/

Теперь кажется, что лучшее решение - просто уметь дизейблить те даты, которые сейчас красным отмечены. То есть иметь функцию как isHoliday, только isDisabled.

**Comment by dzekh** [comment]
Давайте тогда добавим возможность дизейблить отдельные даты. @wKich, норм?

**Comment by wKich** [comment]
Мне не очень нравится вариант появления ещё одной функции `isDisabled`.

**Comment by dzekh** [comment]
Я снимаю тег — договорились что в этом случае подходит стиль задизэйбленных дат. Теперь давайте договоримся как реализовать. 

@wKich чем плохо появление ещё одной функции?

**Comment by wKich** [comment]
Тем что это кривая дорожка к плохому и неудобному API. Сейчас ты их хочешь сделать две, появится третья isHighlighted, потом ещё одна. Представь это как мастер регистрации, где на каждом шаге тебе нужно заполнить ровно одно поле, фамилия, имя и отчество на трёх разных шагах в трёх разных модалках.

**Comment by lossir** [comment]
Если добавить объект с настройками дня, например:
```javascript
date: 'dd.mm.yyyy',
isHoliday: false,
isDisabled: false
...
```
То можно будет оставить только одну функцию перебора, типа `modifierDay`, которая будет принимать эти настройки и возвращать новые.

**Comment by wKich** [comment]
Хорошая мысль

**Comment by wKich** [comment]
Пока не понятно каким цветом красить выделенные даты @dzekh

**Comment by dzekh** [comment]
Нарисовал как могут выглядеть подсвеченные даты: 

<img width="940" alt="Highlighted-dates" src="https://user-images.githubusercontent.com/26383080/84981693-8844d800-b14e-11ea-8a17-e549967e66b9.png">

Точка рядом с датой не мешает выделять даты при наведении и выборе, не сильно бросается в глаза, не мешает воспринимать сетку и не ломает её. Было бы круто если цвет точки можно было бы задать, выбрав из нашего стандартного набора смысловых цветов или красить в цвет бренда.

[Обсуждение в стаффе](https://staff.skbkontur.ru/post/976e9f77b8d3425aaf21e8fd2146e67d).

**Comment by wKich** [comment]
Может линией подчеркивания выделять даты? Набор точек плохо заметен среди кучи чисел.

**Comment by dzekh** [comment]
Попробовал, по мне выглядит странно:
1) если делать толщиной 2px — слишком навязчиво крупно и в глазах рябит.
2) если делать толщиной в 1px — становится похоже на ссылки и вообще странно — хочется уже сделать ссылками — выровнять все линии по ширине с цифрами и покрасить в цвет цифр, что в принципе не правильно.

В Стаффе возражений и предложений других нет, всем ок.

<img width="940" alt="Highlighted-dates-2" src="https://user-images.githubusercontent.com/26383080/85024314-5c961200-b18f-11ea-9a09-5c2ed12ce604.png">

**Comment by wKich** [comment]
Я имел в виду, что соседние даты сплошной линией. Ну если всем норм, то ладно. :)

**Comment by lossir** [comment]
Мне кажется немного перебор с количеством элементов маркировки дня.
У нас есть текст, фон, обводка и появится точка.
![image](https://user-images.githubusercontent.com/2708211/85102083-50f12c80-b21d-11ea-914d-99f913fd10b0.png)
Может попробовать сделать что-то с помощью фона или обводки?
Конечно, главное преимущество нового маркера(точки) в том, что его видно независимо от уже существующих маркеров.

**Comment by slogger** [comment]
В продукте может быть потребность по разному подсвечивать разные дни

В Фокусе у мы хотим показывать изменения некой оценки компании в прошлом и цветом кодировать уровень этой самой оценки (плохая/нормальная/хорошая)
